### PR TITLE
Nano: fix and additions to collection matching, remove cross-cleaning

### PR DESCRIPTION
**Pull request by swertz**
#### PR description:

Fill in a few gaps in the matching between collections in Nano:
- boostedTaus were not used in the matching at all, resulting in the existing branch `boostedTau_jetIdx` to be always `-1`
- add matching between secondary vertices and other objects (only store indices to SVs in the other objects, the SV table is not modified)
- add the following matching cases (matching done by common PF cands) and corresponding index branches:
    - jets with boostedTaus (no boostedTau index branch added to the jets, but `boostedTau_jetIdx` is now usable)
    - jets with SVs (2 indices + nSV stored in jets)
    - lowPtElectrons with photons (photon index stored in lowPtElectrons)
    - taus with electrons (electron index stored in taus)
    - taus with muons (muon index stored in taus)
    - taus with SVs (2 indices + nSV stored in taus)
    - electrons with SVs (SV index stored in electrons)
    - muons with SVs (SV index stored in muons)

Also remove the object `cleanmask` branches because they weren't useful in practice as cross-cleaning is heavily analysis-dependent (depends on the object selection). The cleaning can always be done easily a posteriori using the cross-matching indices stored. The cross-cleaning module is kept in place in case people want to use it in private productions.

NanoAOD file size increase from these changes is 0.02 kb/event.

#### PR validation:

Ran with `runTheMatrix.py --what nano --site "" -l 2500.601`

### CODE REVIEW

**Comment by swertz** [comment]
enable nano

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
+xpog

Changes in Nano expected.

**Comment by swertz** [comment]
I realized late that the matching between lowPtElectrons and jets based on common PFcands didn't work. I also now removed the `cleanmask` variables from the DQM.

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
+1

**Comment by mandrenguyen** [comment]
+reconstruction

**Comment by perrotta** [comment]
@swertz could you please confirm that all changes visible in the comparison tests are understood and consequential to the updates implemented with this PR?
For example, is this removal in the jetPuppiTable ok?
image

**Comment by swertz** [comment]
>  For example, is this removal in the jetPuppiTable ok?

All the other changes (cleanmask removed, new index branches) are expected, but that one looks spurious for a number of reasons:

- 25202.0 is a 2015 workflow that runs nano, while there is not any nano produced for 2015 - we could remove the nano step from that WF
- there should be no puppi jets in Run2 nano at all... but I realize this could happen because we have puppiJets by default in the release and revert to CHS using the `run2_nanoAOD_ANY = run2_nanoAOD_106Xv2` modifier, while 25202.0 (which should really not run nano) only uses the `Run2_2016` era
- in any case, this PR does not touch anything that would cause this...
- the DQM did not flag it: 
image

**Comment by vlimant** [comment]
FYI @slava77 
@perrotta @swertz : checking the file in the output directory for the test, a local version of running 25202.0 with and w/o the PR merged , `Events->Draw("nanoaodFlatTable_jetPuppiTable__NANO.obj.floats_*(\"pt\"!=\"\") >> htemp" )` the distributions are just fine.
One thing I noticed though is that the two first events are swapped, and the "Iteration" mechanism in validate.C https://github.com/cms-sw/cms-bot/blob/master/comparisons/validate.C#L1516 might be fooling the user by not comparing the same event.
workflow 25202.0 could produce NANO in flat output too.

bottom line : there are no changes in Jet_pt (a.k.a "nanoaodFlatTable_jetPuppiTable__NANO.obj.floats_*(\"pt\"!=\"\")

**Comment by vlimant** [comment]
the same will likely show up in #40183 BTW

**Comment by vlimant** [comment]
please test

**Comment by swertz** [comment]
+1

Only expected changes are seen in nano; some other differences are flagged but seem to have the same origin as explained above by @vlimant

**Comment by perrotta** [comment]
@swertz please fix the conflict arisen after the merge of #40209

**Comment by swertz** [comment]
please test

**Comment by swertz** [comment]
For some reason it seems other commits have been included in the test? Obviously the commits in this PR, which were previously validated, have not changed. Would you rather running the tests again @perrotta ?

**Comment by perrotta** [comment]
Please test
(A new IB is now available)

**Comment by swertz** [comment]
+1

(though there are again spurious differences that are entirely unrelated with this PR...)

**Comment by mandrenguyen** [comment]
+reconstruction
These spurious changes are a bit irritating, but looking at the code which is touched there is no way anything but nano is updated.

**Comment by rappoccio** [comment]
ping bot

**Comment by smuzaffar** [comment]
[rappoccio](https://github.com/rappoccio) i think `orp` signatures are still missing for this PR

**Comment by rappoccio** [comment]
@smuzaffar the Jenkins tests are still showing as "pending" and waiting for a test command, which was already done, so I was hoping it would come back to life :)

**Comment by swertz** [comment]
They are shown as pending for #40183, I think it got confused because I opened that one from the same branch...

**Comment by smuzaffar** [comment]
@rappoccio , this pending test `bot/40183/jenkins` is for PR #40183 ( which is close now). Note that the checks belong to the commit, as the same commit is used for #40183 and #40162 that is why you see all the checks in both PRs. As far as this PR is concerned, all checks are passed and you should be able to issue `+1` to merge it

**Comment by rappoccio** [comment]
+1

Jenkins is confused because of the same branch being used for two PRs. (Let's avoid that in the future @swertz )
